src/map/GMap.js

```diff
@@ -305,6 +484,21 @@
             }
             this._prevStreetView = this.streetView();
         }
+        
+        // Enable or disable drawing tools.
+        if (this.drawingTools()) {
+            this._drawingManager.setMap(this._googleMap);
+            
+            // Add drawing complete listener to maintain array of drawingState.
+            google.maps.event.addListener(
+                this._drawingManager,
+                "overlaycomplete",
+                function(){
+                    GMap.prototype.onDrawingComplete.apply(context, arguments);
+                });
+        } else {
+            this._drawingManager.setMap(null);
```

Comment:
I can use the [clearInstanceListeners](https://developers.google.com/maps/documentation/javascript/reference#event) method to just clear any event listeners assigned to the drawing manager object wholesale if that's acceptable here.